include/simulation_time/simulation_clock.hpp

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <cassert>
+#include <chrono>
+#include <stdexcept>
+#include <utility>
+
+namespace ngen {
+
+//! Simulation Clock
+//!
+//! Represents simulation ticks monotonically increasing from zero, and
+//! provides conversion utilities between simulation ticks and
+//! calendar time.
+//!
+//! Implements named requirements: Clock (https://en.cppreference.com/w/cpp/named_req/Clock)
```

Comment:
`Clock` or `TrivialClock`? This disagrees with the PR description.

Reply:
Both, though the intention is to implement `Clock` -- it just happens to also implement `TrivialClock`.

Reply:
I noticed after posting this question that `TrivialClock` is actually the more demanding of the two concepts. I think it's fine to just reference `Clock` in the code, to avoid confusion, though maybe mention it in the PR description.